src/script/lua_api/l_env.cpp

```diff
@@ -359,6 +359,39 @@ int ModApiEnv::l_get_node_or_nil(lua_State *L)
 	return 1;
 }
 
+// bulk_get_node([pos1, pos2, ...])
+// pos = {x=num, y=num, z=num}
+int ModApiEnv::l_bulk_get_node(lua_State *L)
+{
+	GET_ENV_PTR;
+
+	// parameters
+	if (!lua_istable(L, 1)) {
+		return 0;
+	}
+
+	s32 len = lua_objlen(L, 1);
+
+	// Do it
+	std::vector<MapNode> nodes;
+
+	for (s32 i = 1; i <= len; i++) {
+		lua_rawgeti(L, 1, i);
+		nodes.push_back(env->getMap().getNode(read_v3s16(L, -1)));
```

Comment:
it's a bit more complicated but you don't need to store a copy of all the nodes read

Reply:
Yes. But in that case, I will need to push and pop data from the Lua stack simultaneously.
Is there some example of how to do it?

Reply:
```cpp
const int list = lua_gettop(L);
s32 len = lua_objlen(L, 1);

lua_createtable(L, len, 0);
const int result = lua_gettop(L);

auto &map = env->getMap();
for (s32 i = 1; i <= len; i++) {
	lua_rawgeti(L, list, i);
	auto pos = read_v3s16(L, -1);
	lua_pop(L, 1);
	pushnode(map.getNode(pos));
	lua_rawseti(L, result, i);
}

return 1;
```

should work

Reply:
Thanks for your help. Done.